app/services/browser_cache.rb

```diff
@@ -1,10 +1,13 @@
 class BrowserCache
   @cache = LruRedux::Cache.new(1_000)
 
+  # Detects browser attributes from User-Agent, truncated to 2047 bytes due
+  # to: https://github.com/fnando/browser/blob/fa4f685482c315b8/lib/browser/browser.rb#L64-L65
   # @param [String] user_agent
   # @return [Browser]
   def self.parse(user_agent)
-    @cache.getset(user_agent) { Browser.new(user_agent) }
+    return Browser.new(nil) if user_agent.nil?
+    @cache.getset(user_agent) { Browser.new(user_agent.mb_chars.limit(2047).to_s) }
```

Comment:
Could we have used the constant value, to avoid a magic number?

```suggestion
    @cache.getset(user_agent) { Browser.new(user_agent.mb_chars.limit(Browser.user_agent_size_limit - 1).to_s) }
```